checks/doks/snapshots_content_invalid.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2022 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package doks
+
+import (
+	"github.com/digitalocean/clusterlint/checks"
+	"github.com/digitalocean/clusterlint/kube"
+)
+
+func init() {
+	checks.Register(&invalidSnapshotContentCheck{})
+}
+
+type invalidSnapshotContentCheck struct{}
+
+// Name returns a unique name for this check.
+func (i *invalidSnapshotContentCheck) Name() string {
+	return "invalid-volume-snapshot-content"
+}
+
+// Groups returns a list of group names this check should be part of.
+func (i *invalidSnapshotContentCheck) Groups() []string {
+	return []string{"doks"}
+}
+
+// Description returns a detailed human-readable description of what this check
+// does.
+func (i *invalidSnapshotContentCheck) Description() string {
+	return "Checks if there are invalid volume snapshot contents that would fail webhook validation"
+}
+
+// Run runs this check on a set of Kubernetes objects. It can return warnings
+// (low-priority problems) and errors (high-priority problems) as well as an
+// error value indicating that the check failed to run.
+func (i *invalidSnapshotContentCheck) Run(objects *kube.Objects) ([]checks.Diagnostic, error) {
+	var diagnostics []checks.Diagnostic
+	errMsg := "Snapshot content has been marked as invalid by CSI validation - check volumeHandle and snapshotHandle are not both set"
+	ssLabelKey := "snapshot.storage.sigs.k8s.io/invalid-snapshot-content-resource"
+	for _, snapshot := range objects.VolumeSnapshotsV1Content.Items {
+		snapshotLabels := snapshot.Labels
+		if _, ok := snapshotLabels[ssLabelKey]; ok {
+			d := checks.Diagnostic{
+				Severity: checks.Error,
+				Message:  errMsg,
```

Comment:
This will say "Snapshot content" here although it's about the "Snapshot" which might be confusing to users. Let's make sure we call out the right affected resource.

Reply:
This is about the volume snapshot content - it's looping over `objects.VolumeSnapshotsV1Content.Items` and below is looping over `objects.VolumeSnapshotsBetaContent.Items`. I think these error messages are correct?

Reply:
Ah sorry, I misread that.

However, based on [the official example](https://github.com/kubernetes-csi/external-snapshotter/tree/eb447544376eaf7115cfe70bd143a40e7718262a#validating-webhook) it seems like Snapshots (as opposed to SnapshotContents) can also have the invalid label. I understand we need to check those as well (both in v1beta1 and v1)?

Reply:
Yeah this file tests only contents and I have a different test file for snapshots here https://github.com/digitalocean/clusterlint/blob/b344de57979a9fc1f2c18df866e29e3c45e2af06/checks/doks/snapshots_invalid_test.go#L75 I split the tests into `snapshots_content_invalid.go` and `snapshots_invalid.go`